client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -195,19 +201,18 @@ const Checkout = React.createClass( {
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
-		} else if ( cart.create_new_blog && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
-			const domainName = cartItems.getDomainRegistrations( cart )[ 0 ].meta;
+		} else if ( cart.create_new_blog && ! cartItems.hasPlan( cart ) ) {
+			if ( isEmpty( receipt.failed_purchases ) ) {
+				return domainManagementList( map( values( receipt.purchases )[ 0 ], 'meta' )[ 0 ] );
+			}
 
-			return domainManagementList( domainName );
+			return '/start/domain-first';
```

Comment:
For my own understanding, how do we know here that the flow in question is the `domain-first` flow?